daemon/emer-machine-id-provider.c

```diff
@@ -291,24 +296,32 @@ emer_machine_id_provider_get_id (EmerMachineIdProvider *self,
   EmerMachineIdProviderPrivate *priv =
     emer_machine_id_provider_get_instance_private (self);
 
-  static gboolean id_is_valid = FALSE;
-  G_LOCK_DEFINE_STATIC (id_is_valid);
-
-  G_LOCK (id_is_valid);
-  if (!id_is_valid)
+  /* The assumption is that this will not be called
+   * multiple times for the same instance across different
+   * threads. */
```

Comment:
Is this assumption true? Was the lock not there for a reason before?

I believe the answers are “yes; no”; and so I don't think this comment is necessary. There is plenty of non-thread-safe code in this application.

Reply:
Yeah, the lock being there did seem strange (what was more strange was the fact that we were using a static var for this). In any case, I'm almost certain that you're right - just put that comment there in case anyone would pick up on it ;-)